src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -768,6 +771,27 @@ object Parser {
         ghostPrimaryExp |
         operand
 
+    // TODO: change delimiter to { and implement required ambiguity resolution, current: !< X, ..., Z !>
+    // declaredPred!<d1, ..., dn!>
+    lazy val fpredConstruct: Parser[PPredConstructor] =
+      (idnUse ~ predConstructArgs) ^^ { p => PPredConstructor(PFPredBase(p._1), p._2) }
+
+    lazy val mpredConstruct: Parser[PPredConstructor] =
+      ???
```

Comment:
I am not sure that you can use this as an implementation. Wouldn't the parser throw an exception if it is evaluated? You could use the failing parser.

Reply:
I see now,  currently, you use `fpredConsturct` instead of `predConstruct`. That is fine.